design/FY2019/Carroll-MRT.md

```diff
@@ -0,0 +1,49 @@
+Carroll Mean Radiant Temperature option for Interior Radiant Exchange
+=====================================================================
+
+**Neal Kruis, Big Ladder Software, LLC**
+
+## Justification for New Feature ##
+
+A well-known performance bottleneck in EnergyPlus is the calculation of interior long-wave radiation exchange. This is a dense-matrix, linear algebra problem with O(n^2 ) complexity. One approach used in other programs with comparable accuracy is the Carroll method (see Carroll 1980, 1980a, & 1981).
+
+## Approach ##
+
+The Carroll method is an approximation of gray-body long-wave radiation exchange within an enclosure that simplifies the surface-to-surface radiation exchange by using a single, mean radiant temperature node, Tr, that act as a clearinghouse for the radiation heat exchange between surfaces. Instead of solving a dense-matrix, linear algebra problem, the mean radiant temperature can be calculated using a single equation, and subsequently used to determine the net long-wave radiation to/from each surface. Unlike the O(n^2 ) complexity of the current dense-matrix solution, this approach has linear complexity.
```

Comment:
So, this is kind of back to the future. If I'm not mistaken, the MRT method was used in BLAST and DOE2.1E. @RKStrand can confirm. The [comments](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/HeatBalanceIntRadExchange.cc#L96-L99) at the top of `HeatBalanceIntRadExchange.cc` allude to that.

Reply:
Yes, I can confirm that BLAST used the MRT method and so did EnergyPlus early on as it was based on the BLAST heat balance method.  I can't say for DOE-2, but my guess is that it also used MRT or some variation on it.  It doesn't surprise me that this work speeds things up.  The current method for E+ requires potentially a lot of iteration in the interior heat balance.

Reply:
@mjwitte @RKStrand I believe there are several competing MRT methods developed by different people:

- Walton (BLAST)
- Seem (TRNSYS, also known as the star-network)
- Carroll (CSE)

What I've heard (from @chipbarnaby), is that Walton's method does not strictly balance heat, whereas Carroll was able to make this work.

I have yet to read Seem's thesis to see how his approach works.

My understanding is that DOE-2 doesn't explicitly model interior radiation exchange (it uses constant combined convection/radation film factors).

Reply:
It would be good to mention these in the NFP.